Dockerfile

```diff
@@ -0,0 +1,37 @@
+# Use Node Iron LTS (20): https://nodejs.org/en/blog/release/v20.9.0
+FROM node:iron-bullseye-slim as wbs-dev-runner-base
+
+# WBS tests use the Selinium Standalone image so no need for the embedded Chromium
+ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true
+
+# Set the working directory
+WORKDIR /workspace
+
+# hadolint global ignore=DL3059
+SHELL [ "/bin/bash", "-o", "pipefail", "-c" ]
+
+# Install necessary packages except Docker
+RUN apt-get update && \
+    apt-get --no-install-recommends -y install \
+        python3-pip \
+        curl \
+        && ln -sf /usr/bin/python3 /usr/bin/python \
+        && rm -rf /var/lib/apt/lists/* \
+        && pip3 install --no-cache-dir --upgrade \
+            pip \
+            setuptools \
+            requests \
+            bs4 \
+            lxml \
+            black
+# Install Docker CLI
+RUN curl -fsSL https://get.docker.com -o get-docker.sh | bash
+RUN sh get-docker.sh
```

Comment:
```suggestion
# Install necessary packages
RUN apt-get update && \
    apt-get --no-install-recommends -y install \
        curl \
        docker.io \
        python3-requests \
        python3-bs4 \
        python3-lxml \
        black \
        && \
    rm -rf /var/lib/apt/lists/* && \
    ln -sf /usr/bin/python3 /usr/bin/python
```

Reply:
Install everything the debian way, do you remember why we had this get.docker.com thingy?

Reply:
The official Debian packages for docker.io and docker-compose:
```
root@bc5467d1d442:/workspace# docker compose --version
Docker version 20.10.24+dfsg1, build 297e128

root@bc5467d1d442:/workspace# docker-compose --version
docker-compose version 1.29.2, build unknown
```
I'm afraid these are too far back for our purposes.

Reply:
Ok, I see. Actually, debian bookworm not offering docker compose v2 is the real blocker I see here. 

I am still concerned about pulling in stuff from so many different places here...

Reply:
It's a dev container. I welcome any specific amendment you have in mind, but it right now protects us from doing essentially the same thing locally. Yes, we need the latest Docker and that script does just add the apt sources and keys and pulls the latest from the Docker apt repo.